src/pyotp/utils.py

```diff
@@ -73,10 +77,9 @@ def strings_equal(s1, s2):
     except ImportError:
         pass
 
-    if len(s1) != len(s2):
-        return False
-
     differences = 0
-    for c1, c2 in zip(s1, s2):
+    for c1, c2 in izip_longest(s1, s2):
+        if c1 is None or c2 is None:
+            pass
```

Comment:
This line looks odd - `pass` will fallthrough to the next line, and fail when xoring `None`. e.g:

```python
>>> None ^ 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for ^: 'NoneType' and 'int'
```